zerver/lib/message.py

```diff
@@ -484,7 +484,7 @@ def has_message_access(
     stream: Stream | None = None,
     is_subscribed: bool | None = None,
     user_group_membership_details: UserGroupMembershipDetails,
-    is_modifying_message: bool = False,
+    is_modifying_message: bool,
```

Comment:
I think we need to make this change in more of the calling code paths, like `access_message`, to really get the benefit that when reading product code, it's explicit what the request type is.